torchvision/transforms/v2/functional/_geometry.py

```diff
@@ -185,7 +185,35 @@ def resize_image_tensor(
         image = image.reshape(-1, num_channels, old_height, old_width)
 
         dtype = image.dtype
-        need_cast = dtype not in (torch.float32, torch.float64)
+        acceptable_dtypes = [torch.float32, torch.float64]
+        if interpolation == InterpolationMode.NEAREST or interpolation == InterpolationMode.NEAREST_EXACT:
+            # uint8 dtype can be included for cpu and cuda input if nearest mode
+            acceptable_dtypes.append(torch.uint8)
+        elif interpolation == InterpolationMode.BILINEAR and image.device.type == "cpu":
+            # uint8 dtype support for bilinear mode is limited to cpu and
+            # according to our benchmarks non-AVX CPUs should prefer u8->f32->interpolate->u8 path
+            if "AVX2" in torch.backends.cpu.get_cpu_capability():
+                acceptable_dtypes.append(torch.uint8)
+
+                # TODO: Remove when https://github.com/pytorch/pytorch/pull/101136 is landed
+                if dtype == torch.uint8 and not (
+                    image.is_contiguous() or image.is_contiguous(memory_format=torch.channels_last)
+                ):
+                    image = image.contiguous(memory_format=torch.channels_last)
+
+        if image.is_contiguous(memory_format=torch.channels_last):
+            strides = image.stride()
+            numel = image.numel()
+            if image.shape[0] == 1 and numel != strides[0]:
+                # This is the case when channels last tensor was squeezed and unsqueezed such that
+                # stride[0] set as image.shape[1] * image.stride()[1] instead of being image.numel()
+                # Let's restride image such that it will be correctly treated as channels last.
+                # Related pytorch issue: https://github.com/pytorch/pytorch/issues/68430
```

Comment:
I think here it's important to provide details of what we're tying to avoid (an extra copy of the output), and when we should be able to remove those checks. May I suggest the following:

> There is a weird behaviour in torch core where the output tensor of `interpolate()` can be allocated as contiguous even though the input is un-ambiguously channels_last (https://github.com/pytorch/pytorch/issues/68430). In particular this happens for the typical torchvision use-case of single CHW images where we fake the batch dim to become 1CHW. Below, we restride those tensors to trick torch core into properly allocating the output as channels_last, thus preserving the memory format of the input. This is not just for format consistency: for uint8 bilinear images, this also avoids an extra copy (re-packing) of the output and saves time.
TODO: when https://github.com/pytorch/pytorch/issues/68430 is fixed (possibly by https://github.com/pytorch/pytorch/pull/100373), we should be able to remove this hack.